app/assets/javascripts/autocomplete.js

```diff
@@ -1,140 +1,146 @@
+  $.widget("custom.combobox", {
+    _create: function() {
+      this.wrapper = $("<span>")
+        .addClass("constituency-autocomplete")
+        .insertAfter(this.element);
+
+      this.element.hide();
+      this._createAutocomplete();
+      this._createShowAllButton();
+    },
+
+    _createAutocomplete: function() {
+      var element = this.element,
+        selected = element.children(":selected"),
+        value = selected.val()
+          ? selected.text()
+          : "Type to select your constituency";
+
+      var input = (this.input = $("<input>")
+        .appendTo(this.wrapper)
+        .val(value)
+        .attr("title", "")
+        .addClass(
+          "constituency-autocomplete-input ui-widget ui-widget-content ui-state-default ui-corner-left"
+        )
+        .autocomplete({
+          delay: 0,
+          minLength: 0,
+          source: $.proxy(this, "_source"),
+          classes: {
+            "ui-autocomplete": "constituency-autocomplete-select"
+          }
+        })
+        .tooltip({
+          classes: {
+            "ui-tooltip": "ui-state-highlight"
+          }
+        })
+        .focus(function() {
+          var selected = element.children(":selected");
+          if (!selected.val()) {
+            input.val("");
+          }
+        }));
 
-          autocompletechange: "_removeIfInvalid"
-        });
-      },
-
-      _createShowAllButton: function() {
-        var input = this.input,
-          wasOpen = false;
-
-        $( "<a>" )
-          .attr( "tabIndex", -1 )
-          .appendTo( this.wrapper )
-          .button({
-            icons: {
-              primary: "ui-icon-triangle-1-s"
-            },
-            text: false
-          })
-          .removeClass( "ui-corner-all" )
-          .addClass( "constituency-autocomplete-toggle ui-corner-right" )
-          .on( "mousedown", function() {
-            wasOpen = input.autocomplete( "widget" ).is( ":visible" );
-          })
-          .on( "click", function() {
-            input.trigger( "focus" );
-
-            // Close if already visible
-            if ( wasOpen ) {
-              return;
-            }
-
-            // Pass empty string as value to search for, displaying all results
-            input.autocomplete( "search", "" );
+      this._on(this.input, {
+        autocompleteselect: function(event, ui) {
+          ui.item.option.selected = true;
+          this._trigger("select", event, {
+            item: ui.item.option
           });
-      },
+        },
+
+        autocompletechange: "_removeIfInvalid"
+      });
+    },
+
+    _createShowAllButton: function() {
+      var input = this.input,
+        wasOpen = false;
+
+      $("<a>")
+        .attr("tabIndex", -1)
+        .appendTo(this.wrapper)
+        .addClass(
+          "ui-button ui-widget ui-button-icon-only constituency-autocomplete-toggle ui-corner-right"
+        )
+        .append("<span class='ui-button-icon ui-icon ui-icon-triangle-1-s' />")
```

Comment:
This is the key change - `$().button()` is implemented by both BS and jQuery-UI, and BS now wins. We add the button using jQuery manually.

Reply:
Thanks for the explanation!  Tiny nitpick, but in the future I'd recommend putting this stuff in comments in the code, as it will never be found again if only buried in the PR (especially if we want to reap the benefit's of GitHub's "Resolve conversation" button).